scene/gui/popup_menu.cpp

```diff
@@ -1487,6 +1488,7 @@ void PopupMenu::add_icon_radio_check_shortcut(const Ref<Texture2D> &p_icon, cons
 }
 
 void PopupMenu::add_submenu_item(const String &p_label, const String &p_submenu, int p_id) {
+	ERR_FAIL_COND_MSG(p_submenu == SceneStringNames::get_singleton()->dot || p_submenu == SceneStringNames::get_singleton()->doubledot, "Don't use . as submenu's name.");
```

Comment:
You should use `String::validate_node_name()` and then compare the result to the original name and see if it matches. See `SceneTreeEditor::_rename_node` for an example on how to handle it (with a more explicit error message too).

Reply:
Thanks! There are so many good practices I'm not aware of.